### PR TITLE
Create new file

### PR DESCRIPTION
**Pull request by vasubhardwaj011**
- add single space in comment line. [ed960df](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/ed960df763d6b1b005663fc4c50b8a767e5146d6)
- write code to remove last occurrence of character of the string. [6b7c55e](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/6b7c55e80c19778f8aca4a5aaccbda2fd40e1380)
- write code to remove all occurrence of character of the string. [8e4041c](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/8e4041cc10949121183b0cde80b8d738656afabb)
- updated 23_remove_last_occurrence_of_character.js. [4ed0850](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/4ed08502072cb001886bdbf865c311de50f6cbca)
- updated 24_remove_all_occurrence_of_character.js. [11d7b9e](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/11d7b9e8546388c0884779b2f58f564388092cb0)
- update code to avoid inbuilt functions and refactor logic. [cad124d](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/cad124dc773a4196b52e583c21b406c1c6a0b15f)
- update code to avoid inbuilt functions and refactor logic. [f9d0e5a](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/f9d0e5a2200d43f06fe571213f3e9f51fe69e201)]
- add JavaScript program to remove all repeated characters from a string. [759a277](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/759a277e4eaf8495854345af216f273e79df2a62)]

### CODE REVIEW

File: 24_remove_all_occurrence_of_character.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    search: "a",
+    replace_with: "",
+    count: 0
+}
+
+for (let i = 0; i <= text.string.length; i++) {
+
+    if (text.string[i] == text.search) {
+        text.count++;
+        if (text.count == text.count) {
```
Comment (Rohit2820) [inline 24_remove_all_occurrence_of_character.js]: stop drinking in office hours

File: 23_remove_last_occurrence_of_character.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Write a javaScript program to remove last occurrence of a character in a given string.
+
+
+let text = {
+    string: "My Name is Vasu",
+    search: "a",
+    count: 0
```
Comment (Rohit2820) [inline 23_remove_last_occurrence_of_character.js]: what is the use of this

File: 23_remove_last_occurrence_of_character.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+for (let i = 0; i < text.string.length; i++) {
+
+    if (text.string[i] == text.search) {
```
Comment (Rohit2820) [inline 23_remove_last_occurrence_of_character.js]: Told you already, dont drink and code

Comment (vasubhardwaj011) [reply 23_remove_last_occurrence_of_character.js]: I have understood your suggestion and updated this code.
Thank you

**Review by vasubhardwaj011 — COMMENTED** [review]
updated 23_remove_last_occurrence_of_character.js
please approve it.

File: 24_remove_all_occurrence_of_character.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    string: "My Name is Vasuaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+    search: "a",
+    replace_with: ""
+}
+
+for (let i = 0; i < text.string.length; i++) {
+
+    if (text.string[i] == text.search) {
+
+        text.string = text.string.replace(text.string[i], text.replace_with);
```
Comment (kapilPerlStreet) [inline 24_remove_all_occurrence_of_character.js]: @vasubhardwaj011 

Please do not use in-built functions like `replace`.

File: 23_remove_last_occurrence_of_character.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+// let a = text.string.slice(0,text.last_occurrence_char_index);
+// let b = text.string.slice((text.last_occurrence_char_index + 1),text.string.length);
+// text.string = a + b;
+
+//           |
+//          _|_         // Upper code is written at bottom in a single line.
+//          \ /
+//           V
+
+text.string = text.string.slice(0, text.last_occurrence_char_index) + text.string.slice((text.last_occurrence_char_index + 1), text.string.length);
```
Comment (kapilPerlStreet) [inline 23_remove_last_occurrence_of_character.js]: Please do not use in-built functions.

File: 24_remove_all_occurrence_of_character.js (hunk 4 above)
Comment (vasubhardwaj011) [reply 24_remove_all_occurrence_of_character.js]: okk

File: 23_remove_last_occurrence_of_character.js (hunk 5 above)
Comment (vasubhardwaj011) [reply 23_remove_last_occurrence_of_character.js]: okk, I will change it.
Thank you for your review.

**Pull request merged**